src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -790,6 +749,75 @@ public StringBuilder Append(string value, int startIndex, int count)
             }
         }
 
+        public StringBuilder Append(StringBuilder value)
+        {
+            if (value != null && value.Length != 0)
+            {
+                return AppendCore(value, 0, value.Length);
+            }
+            return this;
```

Comment:
if I do sb.Append((StringBuilder)null) shouldn't it throw?

also should `return AppendCore` be outside the if, so it can be the last line and maybe get tailcall optimization (as the common case)

Reply:
> if I do sb.Append((StringBuilder)null) shouldn't it throw?

I was wondering about that as well ...

Reply:
It will return this. similar behavior to append(string)

Reply:
https://github.com/Anipik/coreclr/blob/ee5ecc8be2f16746e975640df381f6fe7a4e580d/src/mscorlib/shared/System/Text/StringBuilder.cs#L690

Reply:
OK, consistency is good. Please make sure there's tests for all these..